### PR TITLE
Multilingual support

### PR DESCRIPTION
**Pull request by mdorf**
Pull request to merge the multilingual enhancements to master. This request implements a dropdown menu to allow switching between available languages when viewing an ontology terms.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -68,9 +68,16 @@ gem 'recaptcha', '~> 5.9.0'
...
 gem 'rexml', '~> 3'
 gem 'stackprof', require: false
+
+# pinning strscan to v 3.0.1 to deal with deployment issue.  Remove line below when issue is fixed
+gem 'strscan', '3.0.1'
+
 gem 'terser'
 gem 'thin'
 gem 'will_paginate', '~> 3.0'
+gem 'net-ftp'
```
Comment (syphax-bouazzouni) [inline Gemfile:78]: not sure `net-ftp` is required

File: app/controllers/language_controller.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class LanguageController < ApplicationController
```
Comment (syphax-bouazzouni) [inline app/controllers/language_controller.rb:1]: this file is not needed

File: app/helpers/internationalisation_helper.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
+module InternationalisationHelper
```
Comment (syphax-bouazzouni) [inline app/helpers/internationalisation_helper.rb:1]: this file is not needed

File: app/helpers/multi_languages_helper.rb [hunk 4]
```diff
@@ -0,0 +1,146 @@
+module MultiLanguagesHelper
+
+  def portal_language_help_text
```
Comment (syphax-bouazzouni) [inline app/helpers/multi_languages_helper.rb:3]: `portal_language_help_text`, `portal_languages` and `portal_language_selector` can be removed as not used

File: app/helpers/multi_languages_helper.rb [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+  end
+
+  def request_lang(submission = @submission || @submission_latest)
+    lang = params[:language] || params[:lang]
+    lang = submission_languages(submission)&.first unless lang
+    lang = portal_lang unless lang
+    lang
+  end
+
+  def lang_code(code_in)
```
Comment (syphax-bouazzouni) [inline app/helpers/multi_languages_helper.rb:127]: this function is not used and can be removed

File: app/views/concepts/_details.html.haml [hunk 6]
```diff
@@ -25,7 +25,11 @@
     %tr
       %td Preferred Name
       %td
-        %p= @concept.prefLabel({:use_html => true}).html_safe
+        %p
+        - if @concept.prefLabel({:use_html => false}).to_s.empty?
```
Comment (syphax-bouazzouni) [inline app/views/concepts/_details.html.haml:29]: can be changed to` @concept.prefLabel.blank?`, to prevent the `nil` exception

**Review by syphax-bouazzouni — COMMENTED** [review]
Added some comments, of not used code that can be removed.
But you adapted well our code to work with the old Jauery code, maybe you should merge it in the `develop` branch ?

File: app/views/concepts/_details.html.haml (hunk 6 above)
Comment (mdorf) [reply app/views/concepts/_details.html.haml:29]: this code `@concept.prefLabel({:use_html => false}).to_s.empty?` will not throw a `nil` exception. 

`nil.to_s` is a valid call that will turn a nil into an empty string. `xxx`.to_s.empty? ` is generally used to check for either `nil` or `empty?` condition.

**Pull request merged**